.github/workflows/ci.yml

```diff
@@ -21,3 +21,28 @@ jobs:
       with:
         source: 'src test'
         extensions: 'ts'
+
+  publish:
+    name: Publish to VSCode marktplace
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'master') # publish it only if the push comes from master.
+    needs: build # make sure the build is passed.
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: detect version
+      uses: EndBug/version-check@v1
+      id: check
+      with:
+        diff-search: true # seach the diff content with version
+        file-name: package.json
+        token: ${{ secrets.GITHUB_TOKEN }}
+    - name: version bumped
```

Comment:
I don't think this step is so useful, the following steps run on the same condition, and it's easy to distinguish steps that are run vs skipped in the github UI